src/primo/browzine-primo-adapter.js

```diff
@@ -705,7 +705,7 @@ browzine.primo = (function() {
         }
       }
 
-      if (request.readyState == XMLHttpRequest.DONE && request.status == 404) {
+      if (request.readyState == XMLHttpRequest.DONE && request.status == 404 || (isArticle(scope) && (!directToPDFUrl && !articleLinkUrl))) {
```

Comment:
What if we added an extra parentheses grouping?

I'm not entirely sure if the order of precedence is either...

```suggestion
      if ((request.readyState == XMLHttpRequest.DONE && request.status == 404) || (isArticle(scope) && (!directToPDFUrl && !articleLinkUrl))) {
```

...or...

```suggestion
      if (request.readyState == XMLHttpRequest.DONE && (request.status == 404 || (isArticle(scope) && (!directToPDFUrl && !articleLinkUrl)))) {
```

Think an extra paren set would be helpful?